### PR TITLE
provider/openstack: register/deregister instances with lbaasv2

### PR DESCRIPTION
**Pull request by drmaas**
Also , startedd on enable/disable server groups (made it compile and roughed it in)

### CODE REVIEW

**Comment by drmaas** [comment]
@derekolk @varikin @spinnaker/reviewers

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy [hunk 1]
```diff
@@ -22,6 +22,7 @@ import org.openstack4j.model.network.ext.HealthMonitorV2
 import org.openstack4j.model.network.ext.LbPoolV2
 import org.openstack4j.model.network.ext.ListenerV2
 import org.openstack4j.model.network.ext.LoadBalancerV2
+import org.openstack4j.model.network.ext.MemberV2
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: Sort-order is off here.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy [hunk 2]
```diff
@@ -190,6 +191,43 @@ interface OpenstackLoadBalancerProvider {
...
   /**
+   * Get port from load balancer listener description. Openstack load balancers have no native concept of internal port,
+   * so we store in the description field of the load balancer.
+   *
+   * This may be changed in a future version.
+   *
+   * @param pool
+   * @return
+   */
+  Integer getInternalLoadBalancerPort(String region, String listenerId)
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: That `@param` lists don't match the method signature for each of these 4 methods.
I'd just drop the `@param` lists entirely (and the `@return`'s) if they're not going to include anything beyond the name of the parameters.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy [hunk 3]
```diff
@@ -27,10 +30,15 @@ import org.openstack4j.model.network.ext.LbPoolV2
 import org.openstack4j.model.network.ext.ListenerProtocol
 import org.openstack4j.model.network.ext.ListenerV2
 import org.openstack4j.model.network.ext.LoadBalancerV2
+import org.openstack4j.model.network.ext.MemberV2
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerV2Provider.groovy]: Sort-order is off here.

**Comment by derekolk** [comment]
LGTM

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy [hunk 4]
```diff
@@ -190,6 +191,44 @@ interface OpenstackLoadBalancerProvider {
...
+
+  /**
+   * Get the member id for instance
+   * @param instanceId
+   * @param ip
+   * @param lbPoolId
+   */
+  String getMemberIdForInstance(String region, String ip, String lbPoolId)
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: `@param` list doesn't match here.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy [hunk 5]
```diff
@@ -190,6 +191,44 @@ interface OpenstackLoadBalancerProvider {
...
+
+  /**
+   * Add a member to a pool.
+   * @param ip
+   * @param lbPoolId
+   * @param subnetId
+   * @param internalPort
+   * @param weight
+   */
+  MemberV2 addMemberToLoadBalancerPool(String region, String ip, String lbPoolId, String subnetId, Integer internalPort, int weight)
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: `@param` list doesn't match here.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy [hunk 6]
```diff
@@ -190,6 +191,44 @@ interface OpenstackLoadBalancerProvider {
...
+
+  /**
+   * Remove a member from a pool.
+   * @param memberId
+   * @return
+   */
+  ActionResponse removeMemberFromLoadBalancerPool(String region, String lbPoolId, String memberId)
```
Comment (duftler) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: `@param` list doesn't match here.

**Comment by duftler** [comment]
LGTM after addressing comments.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy (hunk 5 above)
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackLoadBalancerProvider.groovy]: Addressed.

**Pull request merged**